web/src/channel.ts

```diff
@@ -196,5 +213,5 @@ export function xhr_error_message(message, xhr) {
             message = server_response_html;
         }
     }
-    return message;
+    return message!;
```

Comment:
Maybe an `assert` is more appropriate here?

Reply:
I just checked this again. I think I was wrong. We should change the return type to `string | undefined` instead and remove the assertion. The compose banner (which uses `xhr_error_message` return value) does handle the case when the message is `undefined`. This means that the type annotation of this function also needs to be adjusted: https://github.com/zulip/zulip/blob/ada2991f1c3330c59347b1bdc5ec60b13a71dc24/web/src/compose_banner.ts#L121-L146

Reply:
Oh, thanks for spotting that! Just to be clear this is the one code path which might have `null` message right?

https://github.com/zulip/zulip/blob/ada2991f1c3330c59347b1bdc5ec60b13a71dc24/web/src/message_edit.js#L1034-L1043

Reply:
Yeah, that appears to be the case.